src/engine/renderer/tr_init.cpp

```diff
@@ -74,7 +74,8 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 		Util::ordinal(dynamicLightRenderer_t::LEGACY),
 		Util::ordinal(dynamicLightRenderer_t::TILED) );
 	Cvar::Cvar<bool> r_dynamicLight( "r_dynamicLight", "enable dynamic lights", Cvar::NONE, true );
-	Cvar::Cvar<bool> r_staticLight( "r_staticLight", "enable static lights", Cvar::NONE, true );
+	Cvar::Cvar<bool> r_animatedLight( "r_animatedLight", "enable animated lights (if dynamic light is enabled)", Cvar::NONE, true );
```

Comment:
I find the term "animated light" too random as it does not make any mention of "dynamic" lights. How about making them negative cvars: `r_ignoreDynamicLights` and `r_ignoreLightEntities`? "ignore dynamic lights (even if dynamic light renderer is enabled", "ignore map light entities (even if dynamic light renderer is enabled)"? (Also taking the opportunity to give a better name than 'static light').

Reply:
The “animated” word was simply chosen for consistency with “misc_anim_model” (which is about a non-static model).

I agree the lack of mention of  “dynamic” is confusing, for now I did that to not have to rewrite all the `r_staticLight` stuff, but I can do it. I personally would prefer `r_dynamicAnimatedLights` and `r_dynamicStaticLights`, and if we find a better word than “animated”, I prefer the generic `r_dynamic<Kind>Lights` naming scheme anyway.

Reply:
I meant there are two kind of models: _static_ models to be baked in bsp, and _anim_ models to be loaded in game, so since bsp lights are named _static_, I thought it would not be bad to also name non-bsp lights _anim_, just for consistency of the wording.

The `misc_anim_model` entity kind is not necessarily for using actual animations, it's for any model that is not baked in bsp. If your model needs to be animated, it should be a `misc_anim_model`, if your model needs to move, it better be a `misc_anim_model` (unless you statically bake it into a bsp model that is itself moving), if you want your model to be updated when the game update the model or based on some in-game options (like providing a more detailed model), you want to use `misc_anim_model` even if it is not animated or does not move (a good example are the outdoor decorative turrets in parpax).

On the contrary the static `misc_model` entity is just a kind of prefab that contributes to the bsp (the reason why people may prefer it over `misc_anim_model` is that it then allows to compute lightmaps on them).

Instead of “animatedLights” I was also thinking about “entityLights”, but the BSP is the world entity so it's confusing as well.

Reply:
With that knowledge, if I had to draw an analogy between types of models and lights, I would say the models of players and buildables correspond to dynamic lights, `misc_anim_model` to "static" lights, and `misc_model` to lights baked in lightmaps.

> Instead of “animatedLights” I was also thinking about “entityLights”,

That's funny since I suggested `r_ignoreLightEntities` to replace `r_staticlight`. The static lights can only come from BSP entities, while I believe the cgame can insert dynamic lights any way it wants, independently of any BSP or gamelogic entities.

Reply:
> That's funny since I suggested `r_ignoreLightEntities` to replace `r_staticlight`.

Yeah, that's because you said that I shared my other thought about “entityLights”, I wanted to share the fun! 🤣️

![Confusion, confusion everywhere](https://i.imgflip.com/90c0hq.jpg)

Reply:
Alternative naming can be:

- `cvar bool r_dynamicLight`:  moving lights like particles and weapon projectiles,
- `cvar bool r_staticLight`: map lights (we don't use yet),
- `cvar bool r_realtimeLighting`: enables the rendering of them (currently named `r_dynamicLight`)
- `cvar int r_realtimeLightingRenderer`: select the renderer (currently named `r_dynamicLightRenderer`)

Reply:
I like it

Reply:
I did that.